letter-sender/src/test/java/SendChristmasLetterTest.java

```diff
@@ -0,0 +1,113 @@
+import software.amazon.awssdk.services.sns.model.*;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.*;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class SendChristmasLetterTest {
+
+    public static final String DUMMY = "dummy";
+    private static final String LOCALSTACK_URL = "http://localhost:4566";
+    private static final String LOCALSTACK_ARN = "arn:aws:sqs:us-east-1:000000000000";
+    private static final String REGION = "us-east-1";
+
+    private SnsClient snsClient;
+    private SqsClient sqsClient;
+    private String topicArn;
+    private String queueURL;
+
+    @BeforeAll
+    void setup(){
+        initialiseSnsClient();
+
+        initialiseSqsClient();
+
+        // Creating the SNS Topic
+        CreateTopicResponse topicResponse = snsClient.createTopic(CreateTopicRequest.builder().name("TestTopic").build());
+        topicArn = topicResponse.topicArn();
+
+        // Creating the SQS Queue
+        CreateQueueResponse queueResponse = sqsClient.createQueue(CreateQueueRequest.builder().queueName("TestQueue").build());
+        queueURL = queueResponse.queueUrl();
+
+        //Subscribing the SQS to the SNS
+        snsClient.subscribe(SubscribeRequest.builder()
+                .topicArn(topicArn)
+                .protocol("sqs")
+                .endpoint(LOCALSTACK_ARN + ":TestQueue")
+                .build());
+    }
+
+    void initialiseSnsClient(){
+        snsClient = SnsClient.builder()
+                .endpointOverride(URI.create(LOCALSTACK_URL))
+                .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create(DUMMY, DUMMY)))
+                .region(Region.of(REGION))
+                .build();
+    }
+
+    void initialiseSqsClient(){
+        sqsClient = SqsClient.builder()
+                .endpointOverride(URI.create(LOCALSTACK_URL))
+                .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create(DUMMY, DUMMY)))
+                .region(Region.of(REGION))
+                .build();
+    }
+
+    @Test
+    public void whenSendChristmasLetter_ThenMessageReceivedOnQueue() throws IOException {
+        //Publish message to the SNS Topic
+        String testMessage = "{\n" +
+                "    \"email\": \"luiza.mujdei@gmail.com\",\n" +
+                "    \"name\": \"Luiza\",\n" +
+                "    \"wishes\": \"fimo\",\n" +
+                "    \"location\": \"Iasi\"\n" +
+                "}";
+
+        snsClient.publish(PublishRequest.builder().topicArn(topicArn).message(testMessage).build());
+
+        //Polling the SQS queue
+        ReceiveMessageResponse receiveMessageResponse = sqsClient.receiveMessage(ReceiveMessageRequest.builder()
```

Comment:
Maybe we can simplify this a little bit. If you would post the message as json, maybe you can deserialize it back in a ChristmasLetter instance and compare it to what you were expecting.

Also, what you're testing here is that snsClient knows how to publish a message, and sqsClient knows how to receive it. We should be testing our controller instead. See maybe MockMvc as a solution, example: https://spring.io/guides/gs/testing-web

Reply:
I followed your suggestion and added a ChristmasLetterControllerTest that uses WebMvcTest